File: ws/Perez/U2/WS24/Zoo/src/main/java/ec/edu/espe/zoo/model/Primate.java
package ec.edu.espe.zoo.model;

import java.util.Date;

/**
 *
 * @author Carlos Perez,The Javas, DCCO-ESPE
 */
public abstract class Primate extends Mammal{
    
    public Primate(int id, String description, Date bornOnDate, char gender, Cage cage, int numberOfBones, boolean vetebrate) {
        super(id, description, bornOnDate, gender, cage, numberOfBones, vetebrate);
    }
    
}

File: hw/segarra/u2/HW18ZooInPythonAndJavaScript/ec.edu.zooinjavascript.model/Primate.js
import Mammal from "./Mammal.js";

export default class Primate extends Mammal{
    constructor(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate){
        if (new.target === Primate) {
            throw new TypeError("Cannot construct Primate instances directly");
        }
        super(id, description, bornOnDate, gender, cage, numberOfBones, vertebrate)
    }
}